Update rollback instructions for app versions (#49729)
fleetdm/fleet: articles/fleet-maintained-apps.md

You can also pin via the REST API using the `version` parameter on the [`PATCH /api/v1/fleet/software/titles/:id/package`](https://fleetdm.com/docs/rest-api/rest-api#update-package) endpoint.

## Rollback to a previous version

If a new version introduces a bug, and you want to rollback to the older version, follow the steps below:

> 1. Pin the app to the previous version using the same **Actions > Versions** workflow (or the GitOps `version` key or REST API above).

Reviewer: ```suggestion
1. Pin the app to the previous version. [Learn how](#pin-a-version).
```

I think we can simplify by just pointing to the pin version section w/ instructions.

> 2. If you use [patch policy](https://fleetdm.com/guides/how-to-use-policies-for-patch-management-in-fleet) to keep your app up to date, disable software install automation for this policy.

Author: ```suggestion
2. If you use [patch policy](https://fleetdm.com/guides/how-to-use-policies-for-patch-management-in-fleet) to keep your app up to date, delete the policy.
```

> 2. If you use [patch policy](https://fleetdm.com/guides/how-to-use-policies-for-patch-management-in-fleet) to keep your app up to date, disable software install automation for this policy.

Reviewer: @marko-lisica why not just delete the patch policy? I think that's easier to explain / understand. We're rolling back to we're not patching anymore.

Author: @noahtalerman They will want to keep patching later, but they can always add. Also, once we add a "patch" checkbox, we could just say to uncheck it and check it again when you want to continue patching...

Anyway we can suggest to delete for now.

> 3. Create a new policy that checks if the host has a version with a bug and enable software automation. Use the query below for this policy.

Author: ```suggestion
3. Create a new policy that checks if the host has a version with a bug and enable software automation. Use the query below for this policy.

> Installing an older version over a newer one might cause issues for some apps. Please test it on a smaller scope first.

```

> 3. Create a new policy that checks if the host has a version with a bug and enable software automation. Use the query below for this policy.

Reviewer: @marko-lisica installing the older version over the current version works for most apps?

Meaning, I don't have to uninstall the current version from the host first?

Author: @noahtalerman so far, I just tested Zoom. I downloaded an older version of the package that is offered as FMA, copied the FMA install script to do the same thing as Fleet would do, and it worked. I also updated to the latest version, which also worked.

Author: I guess we should probably add a callout to test this on a single host to make sure the install over an existing installation works as expected.

Reviewer: > callout to test this on a single host to make sure the install over an existing installation works as expected

Good call 👍

```sql
SELECT 1 WHERE NOT EXISTS (
    SELECT 1 FROM programs WHERE name = 'Zoom' AND version = '<version_with_bug>'
);

> ```

Author: @cdcme, can you please double-check if this is true:

<img width="764" height="344" alt="Image" src="https://github.com/user-attachments/assets/d0e6b0b0-8789-41fc-99fb-c2ba2f15d5ef" />

Reviewer 2: @marko-lisica This is right 👍

Author: Ty!

You can also pin via the REST API using the `version` parameter on the [`PATCH /api/v1/fleet/software/titles/:id/package`](https://fleetdm.com/docs/rest-api/rest-api#update-package) endpoint.

## Rollback to a previous version

If a new version introduces a bug, and you want to rollback to the older version, follow the steps below:

1. Pin the app to the previous version. [Learn how](#pin-a-version).
2. If you use [patch policy](https://fleetdm.com/guides/how-to-use-policies-for-patch-management-in-fleet) to keep your app up to date, delete the policy.
3. Create a new policy that checks if the host has a version with a bug and enable software automation. Use the query below for this policy.

> Installing an older version over a newer one might cause issues for some apps. Please test it on a smaller scope first.

> 

Author: ```suggestion
```
